GVFS/GVFS.FunctionalTests/Tests/EnlistmentPerFixture/GitBlockCommandsTests.cs

```diff
@@ -0,0 +1,106 @@
+﻿using GVFS.FunctionalTests.FileSystemRunners;
+using GVFS.FunctionalTests.Should;
+using GVFS.FunctionalTests.Tools;
+using GVFS.Tests.Should;
+using NUnit.Framework;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Runtime.InteropServices;
+
+namespace GVFS.FunctionalTests.Tests.EnlistmentPerFixture
+{
+    [TestFixtureSource(typeof(FileSystemRunner), nameof(FileSystemRunner.Runners))]
+    [Category(Categories.GitCommands)]
+    public class GitBlockCommandsTests : TestsWithEnlistmentPerFixture
+    {
+        private FileSystemRunner fileSystem;
+        public GitBlockCommandsTests(FileSystemRunner fileSystem)
+        {
+            this.fileSystem = fileSystem;
+        }
+
+        [TestCase, Order(1)]
+        public void GitFsck()
+        {
+            ProcessResult result = GitHelpers.InvokeGitAgainstGVFSRepo(
+                this.Enlistment.RepoRoot,
+                "fsck");
+            result.ExitCode.ShouldNotEqual(0, result.Errors);

```

Comment:
I would maybe add a function that is:
```
private void CommandBlocked(string command)
{
	ProcessResult result = GitHelpers.InvokeGitAgainstGVFSRepo(
		this.Enlistment.RepoRoot,
		command);
	result.ExitCode.ShouldNotEqual(0, $"Command {command} not blocked when it should be.  Errors: {result.Errors}");
}
```
And have a similar method for `CommandNotBlocked`
Then you could have one test case that is:
```
CommandBlocked("fsck");
CommandBlocked("gc");
CommandNotBlocked("gc --auto");
CommandBlocked("prune");
etc...
```